Userland/Libraries/LibVideo/MatroskaReader.cpp

```diff
@@ -264,6 +277,51 @@ OwnPtr<TrackEntry> MatroskaReader::parse_track_entry()
     return track_entry;
 }
 
+Optional<TrackEntry::ColorFormat> MatroskaReader::parse_video_color_information()
+{
+    TrackEntry::ColorFormat color_format {};
+
+    auto success = parse_master_element("Colour"sv, [&](u64 element_id) {
+        switch (element_id) {
+        case PRIMARIES_ID: {
+            auto primaries_result = read_u64_element();
+            CHECK_HAS_VALUE(primaries_result);
```

Comment:
All these optional guys feel kinda strange. 

Most of the lowest level functions in the library, such as `read_u64_element` have known failure patterns like "out of data" or "expected X bytes" or some kind of verbose dbgln_if that expose some richer context on the error. 

I would consider refactoring them to use your fancy Error type to yeet that information up the stack rather than chaining optionals like you are here with the CHECK_HAS_VALUE macro.

Reply:
I was planning on doing that at some point, but someone got to it before I could, it's in progress actually. I certainly wasn't happy with all the Optionals either lol